### PR TITLE
Add Button-chaser 

### PR DESCRIPTION
**Pull request by Koch-Dustin**
Changes:
   - Added Version1
   - Added Version2
   - Added Version3
   
PR: #1

### CODE REVIEW

File: Version3/index.js [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+})
+
+div.addEventListener('mousemove', (e) => {
+  const x = e.pageX
+  const y = e.pageY
+  const divBox = div.getBoundingClientRect()
+  const horizontalDistanceFrom = distanceFromCenter(divBox.x, x, divBox.width)
+  const verticalDistanceFrom = distanceFromCenter(divBox.y, y, divBox.height)
+  Distancey = verticalDistanceFrom
+  Distancex = horizontalDistanceFrom
```
Comment (stefanluth) [inline Version3/index.js:16]: Why do you remap this variable? :)

File: Version3/index.js [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+
+  if (Distancex != 0) {
+    if (Distancex > 0)
+      pushx = divBox.x + 10;
```
Comment (stefanluth) [inline Version3/index.js:20]: camelCase and don't forget to use `let` or `const` :)

**Review by stefanluth — CHANGES_REQUESTED** [review]

File: Version1/index.html [hunk 3]
```diff
@@ -0,0 +1,13 @@
+<!DOCTYPE html>
+
+<html>
+
+    <head>
```
Comment (DevBastiS) [inline Version1/index.html:5]: Indent in HTML files should always be 2 spaces. Please adjust your settings.

File: Version1/index.js [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+    top = window.innerHeight / 2
+  }
+
+  button.style.left = `${left}px`
+  button.style.top = `${top}px`
+}
+
+function distanceFromCenter(boxPosition, mousePosition, boxSize) {
+  return boxPosition - mousePosition + boxSize / 2
+}
```
Comment (DevBastiS) [inline Version1/index.js:53]: Always add a new line to the end of the file.

File: Version1/index.js [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
+  if (Math.abs(horizontalDistanceFrom) <= horizontalOffset && Math.abs(verticalDistanceFrom) <= verticalOffset) {
+    setButtonPosition( buttonBox.x + horizontalOffset / horizontalDistanceFrom * 10, buttonBox.y + verticalOffset / verticalDistanceFrom * 10)
+  }
+})
+
+function setButtonPosition(left, top) {
+  const windowBox = document.body.getBoundingClientRect()
+  const buttonBox = button.getBoundingClientRect()
+
+  if(distanceFromCenter(left, windowBox.left, buttonBox.width) < 0) {
```
Comment (DevBastiS) [inline Version1/index.js:27]: Move this if conditions into speaking variables.

File: Version2/index.js [hunk 6]
```diff
@@ -0,0 +1,89 @@
+let button = document.getElementById('button');
+const OFFSET = 100
+
+left = button.style.left;
+top = button.style.top;
+
+button.addEventListener('mousemove', function() {
+        button.style.left = "250px"
```
Comment (DevBastiS) [inline Version2/index.js:8]: Weird indent here

File: Version2/index.js [hunk 7]
```diff
@@ -0,0 +1,89 @@
...
+        button.style.top = "70px"
+})
+
+document.addEventListener('mousemove', (e) => {
+    const x = e.pageX
```
Comment (DevBastiS) [inline Version2/index.js:13]: Indent in js files should always be 2 spaces

File: Version2/index.js [hunk 8]
```diff
@@ -0,0 +1,89 @@
...
+    const y = e.pageY
+    const buttonBox = button.getBoundingClientRect()
+    const horizontalDistanceFrom = distanceFromCenter(buttonBox.x, x, buttonBox.width)
+    const verticalDistanceFrom = distanceFromCenter(buttonBox.y, y, buttonBox.height)
+
+    button_from_right = (horizontalDistanceFrom >= -145 && horizontalDistanceFrom <= -45) && (verticalDistanceFrom <= 20 && verticalDistanceFrom >= -25);
```
Comment (DevBastiS) [inline Version2/index.js:19]: Use camelcase in JS. Also what does "buttom from right" mean?

File: Version2/style.css [hunk 9]
```diff
@@ -0,0 +1,22 @@
+*{
```
Comment (DevBastiS) [inline Version2/style.css:1]:
```suggestion
* {
```

File: Version2/style.css [hunk 10]
```diff
@@ -0,0 +1,22 @@
...
+    user-select: none;
```
Comment (DevBastiS) [inline Version2/style.css:2]: Indent in CSS files should always be 2 spaces.

File: Version3/index.js [hunk 11]
```diff
@@ -0,0 +1,57 @@
...
+
+  if (Distancey != 0) {
+    if (Distancey > 0)
+      pushy = divBox.y + 10;
+    else 
+      pushy = divBox.y - 10;
+    div.style.top = `${pushy}px`;
+  }
+
+  if(divBox.x < -120) {
```
Comment (DevBastiS) [inline Version3/index.js:34]: Move if condition to speaking variable.

Comment (DevBastiS) [inline Version3/index.js:34]: Also what does 120 mean?

**Review by DevBastiS — CHANGES_REQUESTED** [review]

File: Version1/index.js [hunk 12]
```diff
@@ -0,0 +1,58 @@
...
+  const y = e.pageY
+  const buttonBox = button.getBoundingClientRect()
+  const horizontalDistanceFrom = distanceFromCenter(buttonBox.x, x, buttonBox.width)
+  const verticalDistanceFrom = distanceFromCenter(buttonBox.y, y, buttonBox.height)
+  const horizontalOffset = buttonBox.width / 2 + OFFSET
+  const verticalOffset = buttonBox.height / 2 + OFFSET
+
+  mouseHitsVerticalTriggerpoint = Math.abs(horizontalDistanceFrom) <= horizontalOffset && Math.abs(verticalDistanceFrom) <= verticalOffset
+  if (mouseHitsVerticalTriggerpoint) {
+    setButtonPosition( buttonBox.x + horizontalOffset / horizontalDistanceFrom * 10, buttonBox.y + verticalOffset / verticalDistanceFrom * 10)
```
Comment (DevBastiS) [inline Version1/index.js:20]:
```suggestion
    setButtonPosition(buttonBox.x + horizontalOffset / horizontalDistanceFrom * 10, buttonBox.y + verticalOffset / verticalDistanceFrom * 10)
```

File: Version2/index.js [hunk 13]
```diff
@@ -0,0 +1,128 @@
...
+
+button.addEventListener("mousemove", function () {
+  button.style.left = "250px";
+  button.style.top = "70px";
+});
+
+document.addEventListener("mousemove", (e) => {
+  const x = e.pageX;
+  const y = e.pageY;
+  const buttonBox = button.getBoundingClientRect();
```
Comment (DevBastiS) [inline Version2/index.js:15]:
```suggestion
  const buttonBox = button.getBoundingClientRect();
  
```

File: Version2/index.js [hunk 14]
```diff
@@ -0,0 +1,128 @@
+let button = document.getElementById("button");
```
Comment (DevBastiS) [inline Version2/index.js:1]:
```suggestion
let buttonToChase = document.getElementById("button");
```

File: Version2/index.js [hunk 15]
```diff
@@ -0,0 +1,128 @@
...
+button.addEventListener("mousemove", function () {
+  button.style.left = "250px";
+  button.style.top = "70px";
+});
+
+document.addEventListener("mousemove", (e) => {
+  const x = e.pageX;
+  const y = e.pageY;
+  const buttonBox = button.getBoundingClientRect();
+  const horizontalDistanceFrom = distanceFromCenter(
```
Comment (DevBastiS) [inline Version2/index.js:16]:
```suggestion
  const horizontalDistanceFromButton = distanceFromCenter(
```

File: Version2/index.js [hunk 16]
```diff
@@ -0,0 +1,128 @@
...
+    buttonBox.x,
+    x,
+    buttonBox.width
+  );
+  const verticalDistanceFrom = distanceFromCenter(
```
Comment (DevBastiS) [inline Version2/index.js:21]:
```suggestion
  const verticalDistanceFromButton = distanceFromCenter(
```

File: Version2/index.js [hunk 17]
```diff
@@ -0,0 +1,128 @@
...
+    buttonBox.y,
+    y,
+    buttonBox.height
+  );
+
+  mouse_from_right =
```
Comment (DevBastiS) [inline Version2/index.js:27]:
```suggestion
  const mouseComesFromRightSide =
```

File: Version2/index.js [hunk 18]
```diff
@@ -0,0 +1,128 @@
...
+    horizontalDistanceFrom >= -145 &&
+    horizontalDistanceFrom <= -45 &&
+    verticalDistanceFrom <= 20 &&
+    verticalDistanceFrom >= -25;
+  if (mouse_from_right) {
+    console.log("Rechts");
+    set_button_position(buttonBox.x - 100 * 6);
+  }
+
+  mouse_from_left =
```
Comment (DevBastiS) [inline Version2/index.js:37]: Same as above.

File: Version2/index.js [hunk 19]
```diff
@@ -0,0 +1,128 @@
...
+    horizontalDistanceFrom <= 145 &&
+    horizontalDistanceFrom >= 45 &&
+    verticalDistanceFrom <= 20 &&
+    verticalDistanceFrom >= -25;
+  if (mouse_from_left) {
+    console.log("Links");
+    set_button_position(buttonBox.x - 100 * 5);
+  }
+
+  mouse_from_top =
```
Comment (DevBastiS) [inline Version2/index.js:47]: Same here

File: Version2/index.js [hunk 20]
```diff
@@ -0,0 +1,128 @@
...
+    horizontalDistanceFrom >= -70 &&
+    horizontalDistanceFrom <= 67 &&
+    verticalDistanceFrom <= 72 &&
+    verticalDistanceFrom >= 18;
+  if (mouse_from_top) {
+    console.log("Oben");
+    set_button_position(0, buttonBox.y - 100 * 3);
+  }
+
+  mouse_from_bottom =
```
Comment (DevBastiS) [inline Version2/index.js:57]: Same here

File: Version2/index.js [hunk 21]
```diff
@@ -0,0 +1,128 @@
...
+    set_button_position(buttonBox.x - 100 * 5);
+  }
+
+  mouse_from_top =
+    horizontalDistanceFrom >= -70 &&
+    horizontalDistanceFrom <= 67 &&
+    verticalDistanceFrom <= 72 &&
+    verticalDistanceFrom >= 18;
+  if (mouse_from_top) {
+    console.log("Oben");
```
Comment (DevBastiS) [inline Version2/index.js:53]: remove console logs.

File: Version2/index.js [hunk 22]
```diff
@@ -0,0 +1,128 @@
...
+  mouse_from_bottom =
+    horizontalDistanceFrom >= -70 &&
+    horizontalDistanceFrom <= 67 &&
+    verticalDistanceFrom >= -72 &&
+    verticalDistanceFrom <= -22;
+  if (mouse_from_bottom) {
+    set_button_position(0, buttonBox.y - 100 * 4);
+  }
+
+  mouse_from_top_right =
```
Comment (DevBastiS) [inline Version2/index.js:66]: Same here

File: Version2/index.js [hunk 23]
```diff
@@ -0,0 +1,128 @@
...
+    horizontalDistanceFrom <= -70 &&
+    horizontalDistanceFrom >= -145 &&
+    verticalDistanceFrom <= 70 &&
+    verticalDistanceFrom >= 20;
+  if (mouse_from_top_right) {
+    set_button_position(buttonBox.x - 100 * 6, buttonBox.y - 100 * 3);
+  }
+
+  mouse_from_bottom_right =
```
Comment (DevBastiS) [inline Version2/index.js:75]: Same here

File: Version2/index.js [hunk 24]
```diff
@@ -0,0 +1,128 @@
...
+    horizontalDistanceFrom <= -70 &&
+    horizontalDistanceFrom >= -145 &&
+    verticalDistanceFrom >= -70 &&
+    verticalDistanceFrom <= -25;
+  if (mouse_from_bottom_right) {
+    set_button_position(buttonBox.x - 100 * 6, buttonBox.y - 100 * 4);
+  }
+
+  mouse_from_bottom_left =
```
Comment (DevBastiS) [inline Version2/index.js:84]: Same here

File: Version2/index.js [hunk 25]
```diff
@@ -0,0 +1,128 @@
...
+    horizontalDistanceFrom >= 66 &&
+    horizontalDistanceFrom <= 144 &&
+    verticalDistanceFrom >= -70 &&
+    verticalDistanceFrom <= -25;
+  if (mouse_from_bottom_left) {
+    set_button_position(buttonBox.x - 100 * 5, buttonBox.y - 100 * 4);
+  }
+
+  mouse_from_top_left =
```
Comment (DevBastiS) [inline Version2/index.js:93]: You get the point ;)

File: Version2/index.js [hunk 26]
```diff
@@ -0,0 +1,128 @@
...
+    verticalDistanceFrom <= 70 &&
+    verticalDistanceFrom >= 20;
+  if (mouse_from_top_left) {
+    set_button_position(buttonBox.x - 100 * 5, buttonBox.y - 100 * 3);
+  }
+
+  let xborder = button.getBoundingClientRect().x;
+  let yborder = button.getBoundingClientRect().y;
+
+  if (xborder <= 0 || xborder >= 1660) {
```
Comment (DevBastiS) [inline Version2/index.js:105]: Move condition to speaking variable.

File: Version2/index.js [hunk 27]
```diff
@@ -0,0 +1,128 @@
...
+
+  console.log(yborder);
+
+  if (yborder <= 7 || yborder >= 900) {
+    button.style.left = "250px";
+    button.style.top = "70px";
+  }
+});
+
+function set_button_position(left, top) {
```
Comment (DevBastiS) [inline Version2/index.js:118]:
```suggestion
function setButtonPosition(left, top) {
```

Sometimes you do it right, sometimes not.

File: Version2/index.js [hunk 28]
```diff
@@ -0,0 +1,128 @@
...
+  button.style.left = `${left}px`;
+  button.style.top = `${top}px`;
+
+  left = left;
+  top = top;
+}
+
+function distanceFromCenter(boxPosition, mousePosition, boxSize) {
```
Comment (DevBastiS) [inline Version2/index.js:126]: Here it is correct

**Review by DevBastiS — CHANGES_REQUESTED** [review]

File: Version1/index.js [hunk 29]
```diff
@@ -0,0 +1,58 @@
...
+  }
+})
+
+function setButtonPosition(left, top) {
+  const windowBox = document.body.getBoundingClientRect()
+  const buttonBox = button.getBoundingClientRect()
+
+  buttonHitsLeftEdge = distanceFromCenter(left, windowBox.left, buttonBox.width) < 0
+  if(buttonHitsLeftEdge) {
```
Comment (stefanluth) [inline Version1/index.js:29]:
```suggestion
  if (buttonHitsLeftEdge) {
```

And other occurrences :)

File: Version1/index.js [hunk 30]
```diff
@@ -0,0 +1,58 @@
...
+  }
+
+  buttonHitsRightEdge = distanceFromCenter(left, windowBox.right, buttonBox.width) > 0
+  if(buttonHitsRightEdge) {
+    left = window.innerWidth / 2
+    top = window.innerHeight / 2
+  }
+
+  buttonHitsUpperEdge = distanceFromCenter(top, windowBox.top, buttonBox.height) < 0
+  if(ButtonHitsUpperEdge) {
```
Comment (stefanluth) [inline Version1/index.js:41]:
```suggestion
  if(buttonHitsUpperEdge) {
```

**Review by stefanluth — CHANGES_REQUESTED** [review]

File: Version1/index.js [hunk 31]
```diff
@@ -0,0 +1,58 @@
...
+  if (buttonHitsUpperEdge) {
+    left = window.innerWidth / 2
+    top = window.innerHeight / 2
+  }
+  
+  buttonHitsLowerEdge = distanceFromCenter(top, windowBox.bottom, buttonBox.height) > 0
+  if (buttonHitsLowerEdge) {
+    left = window.innerWidth / 2
+    top = window.innerHeight / 2
+  }
```
Comment (DevBastiS) [inline Version1/index.js:50]:
```suggestion
  buttonHitsLeftEdge = distanceFromCenter(left, windowBox.left, buttonBox.width) < 0;
  buttonHitsRightEdge = distanceFromCenter(left, windowBox.right, buttonBox.width) > 0;
  buttonHitsUpperEdge = distanceFromCenter(top, windowBox.top, buttonBox.height) < 0;
  buttonHitsLowerEdge = distanceFromCenter(top, windowBox.bottom, buttonBox.height) > 0;
  buttonHitsAnyEdge = buttonHitsLeftEdge || buttonHitsRightEdge || buttonHitsUpperEdge || buttonHitsLowerEdge
  
  if (buttonHitsAnyEdge) {
    left = window.innerWidth / 2;
    top = window.innerHeight / 2;
    
    return;
  }
```

File: Version2/index.js [hunk 32]
```diff
@@ -0,0 +1,126 @@
...
+  const ButtonHitsTheEndOfTheBrowserOnTheXCoordinate = xborder <= 0 || xborder >= 1660
+  if (ButtonHitsTheEndOfTheBrowserOnTheXCoordinate) {
+    buttonToChase.style.left = "250px";
+    buttonToChase.style.top = "70px";
+  }
+
+  const ButtonHitsTheEndOfTheBrowserOnTheYCoordinate = yborder <= 7 || yborder >= 900
+  if (ButtonHitsTheEndOfTheBrowserOnTheYCoordinate) {
+    buttonToChase.style.left = "250px";
+    buttonToChase.style.top = "70px";
```
Comment (DevBastiS) [inline Version2/index.js:112]: Same refactoring possible as the one above.

File: Version3/index.js [hunk 33]
```diff
@@ -0,0 +1,65 @@
...
+  if(buttonHitsUpperEdge) {
+    div.style.left = "40%"
+    div.style.top = "30%"
+  }
+
+  buttonHitsLowerEdge = divBox.y > lowerEdge
+  if(buttonHitsLowerEdge) {
+    div.style.left = "400px"
+    div.style.top = "400px"
+  }
```
Comment (DevBastiS) [inline Version3/index.js:60]: Same refactoring possible.

**Review by DevBastiS — CHANGES_REQUESTED** [review]

File: Version1/index.js [hunk 34]
```diff
@@ -0,0 +1,58 @@
...
+  alert('You won')
+  window.close()
+})
+
+document.addEventListener('mousemove', (e) => {
+  const x = e.pageX
+  const y = e.pageY
+  const buttonBox = button.getBoundingClientRect()
+  const horizontalDistanceFrom = distanceFromCenter(buttonBox.x, x, buttonBox.width)
+  const verticalDistanceFrom = distanceFromCenter(buttonBox.y, y, buttonBox.height)
```
Comment (stefanluth) [inline Version1/index.js:14]: distance from what? :D

File: Version1/style.css [hunk 35]
```diff
@@ -0,0 +1,16 @@
...
+
+#button {
+    position: absolute;
+    white-space: nowrap;
+    font-size: 2rem;
+    border: 1px solid black;
+    border-radius: 5px;
+    background: none;
+    box-shadow: 10px 5px 5px black;
+}
```
Comment (stefanluth) [inline Version1/style.css:16]: Two spaces indent :)

File: Version2/index.js [hunk 36]
```diff
@@ -0,0 +1,126 @@
...
+    buttonBox.y,
+    y,
+    buttonBox.height
+  );
+
+  const mouseComesFromRightSide =
+    horizontalDistanceFromButton >= -145 &&
+    horizontalDistanceFromButton <= -45 &&
+    verticalDistanceFromButton <= 20 &&
+    verticalDistanceFromButton >= -25;
```
Comment (stefanluth) [inline Version2/index.js:32]: The values seem arbitrary. Add variables :)

**Review by stefanluth — CHANGES_REQUESTED** [review]

File: Version1/index.js [hunk 37]
```diff
@@ -0,0 +1,69 @@
...
+  );
+  const verticalDistanceFromButtonCenter = distanceFromCenter(
+    buttonBox.y,
+    y,
+    buttonBox.height
+  );
+  const horizontalOffset = buttonBox.width / 2 + OFFSET;
+  const verticalOffset = buttonBox.height / 2 + OFFSET;
+
+  mouseHitsVerticalTriggerpoint =
```
Comment (stefanluth) [inline Version1/index.js:26]:
```suggestion
  const mouseHitsVerticalTriggerpoint =
```

File: Version1/index.js [hunk 38]
```diff
@@ -0,0 +1,69 @@
...
+
+  buttonHitsLeftEdge =
+    distanceFromCenter(left, windowBox.left, buttonBox.width) < 0;
+  buttonHitsRightEdge =
+    distanceFromCenter(left, windowBox.right, buttonBox.width) > 0;
+  buttonHitsUpperEdge =
+    distanceFromCenter(top, windowBox.top, buttonBox.height) < 0;
+  buttonHitsLowerEdge =
+    distanceFromCenter(top, windowBox.bottom, buttonBox.height) > 0;
+  buttonHitsAnyEdge =
```
Comment (stefanluth) [inline Version1/index.js:50]: Add `const`s

File: Version1/index.js [hunk 39]
```diff
@@ -0,0 +1,69 @@
...
+    top = window.innerHeight / 2;
+
+    return;
+  }
+
+  button.style.left = `${left}px`;
+  button.style.top = `${top}px`;
+}
+
+function distanceFromCenter(boxPosition, mousePosition, boxSize) {
```
Comment (stefanluth) [inline Version1/index.js:67]:
```suggestion
function getDistanceFromCenter(boxPosition, mousePosition, boxSize) {
```

or

```suggestion
function calculateDistanceFromCenter(boxPosition, mousePosition, boxSize) {
```

File: Version2/index.js [hunk 40]
```diff
@@ -0,0 +1,166 @@
+let buttonToChase = document.getElementById("button");
+const OFFSET = 100;
+
+left = buttonToChase.style.left;
+top = buttonToChase.style.top;
```
Comment (stefanluth) [inline Version2/index.js:5]: `const`

File: Version2/index.js [hunk 41]
```diff
@@ -0,0 +1,166 @@
...
+  );
+  const verticalDistanceFromButton = distanceFromCenter(
+    buttonBox.y,
+    y,
+    buttonBox.height
+  );
+  const RightSideRightBorderCoordinateValue = -145;
+  const RightSideLeftBorderCoordinateValue = -45;
+  const RideSideUpperBorderCoordinateValue = -25;
+  const RideSideLowerBorderCoordinateValue = 20;
```
Comment (stefanluth) [inline Version2/index.js:30]: camelCase :)

Comment (stefanluth) [reply Version2/index.js:30]: For the lines below as well. Or actually: You should declare constants like these at the beginning of the file, as `UPPER_CASE_CONSTANTS`.

File: Version2/index.js [hunk 42]
```diff
@@ -0,0 +1,166 @@
...
+
+  let xborder = buttonToChase.getBoundingClientRect().x;
+  let yborder = buttonToChase.getBoundingClientRect().y;
+
+  const ButtonHitsTheEndOfTheBrowserOnTheXCoordinate =
+    xborder <= 0 || xborder >= 1660;
+  const ButtonHitsTheEndOfTheBrowserOnTheYCoordinate =
+    yborder <= 7 || yborder >= 900;
+
+  ButtonHitsAnyEndOfTheBrowser =
```
Comment (stefanluth) [inline Version2/index.js:147]: `const`

File: Version2/index.js [hunk 43]
```diff
@@ -0,0 +1,166 @@
...
+    buttonToChase.style.top = "70px";
+  }
+});
+
+function setButtonPosition(left, top) {
+  buttonToChase.style.left = `${left}px`;
+  buttonToChase.style.top = `${top}px`;
+
+  left = left;
+  top = top;
```
Comment (stefanluth) [inline Version2/index.js:161]: This doesn't do anything :)

File: Version2/index.js [hunk 44]
```diff
@@ -0,0 +1,166 @@
...
+}
+
+function distanceFromCenter(boxPosition, mousePosition, boxSize) {
```
Comment (stefanluth) [inline Version2/index.js:164]: add `get` or `calculate` to function name

File: Version3/index.js [hunk 45]
```diff
@@ -0,0 +1,52 @@
...
+      pushY = divBox.y - 10;
+    div.style.top = `${pushY}px`;
+  }
+
+  buttonHitsLeftEdge = divBox.x < leftEdge
+  buttonHitsRightEdge = divBox.x > rightEdge;
+  buttonHitsUpperEdge = divBox.y < upperEdge;
+  buttonHitsLowerEdge = divBox.y > lowerEdge
+
+  buttonHitsAnyEdge = buttonHitsLeftEdge || buttonHitsRightEdge || buttonHitsUpperEdge || buttonHitsLowerEdge
```
Comment (stefanluth) [inline Version3/index.js:43]: `const`

File: Version3/index.js [hunk 46]
```diff
@@ -0,0 +1,52 @@
...
+  if(buttonHitsAnyEdge) {
```
Comment (stefanluth) [inline Version3/index.js:44]:
```suggestion
  if (buttonHitsAnyEdge) {
```

**Review by stefanluth — CHANGES_REQUESTED** [review]

File: Version1/index.js [hunk 47]
```diff
@@ -0,0 +1,69 @@
...
+  const horizontalOffset = buttonBox.width / 2 + OFFSET;
+  const verticalOffset = buttonBox.height / 2 + OFFSET;
+
+  mouseHitsVerticalTriggerpoint =
+    Math.abs(horizontalDistanceFromButtonCenter) <= horizontalOffset &&
+    Math.abs(verticalDistanceFromButtonCenter) <= verticalOffset;
+  if (mouseHitsVerticalTriggerpoint) {
+    setButtonPosition(
+      buttonBox.x +
+        (horizontalOffset / horizontalDistanceFromButtonCenter) * 10,
```
Comment (DevBastiS) [inline Version1/index.js:32]: Add a new speaking variable for that.

File: Version1/index.js [hunk 48]
```diff
@@ -0,0 +1,69 @@
...
+      buttonBox.y + (verticalOffset / verticalDistanceFromButtonCenter) * 10
```
Comment (DevBastiS) [inline Version1/index.js:33]: For this too, try to apply that you only use 1 allocation in 1 line of code.

File: Version2/index.js [hunk 49]
```diff
@@ -0,0 +1,166 @@
...
+  const mouseComesFromRightSide =
+    horizontalDistanceFromButton >= RightSideRightBorderCoordinateValue &&
+    horizontalDistanceFromButton <= -RightSideLeftBorderCoordinateValue &&
+    verticalDistanceFromButton <= RideSideLowerBorderCoordinateValue &&
+    verticalDistanceFromButton >= -RideSideUpperBorderCoordinateValue;
+  if (mouseComesFromRightSide) {
+    setButtonPosition(buttonBox.x - 100 * 6);
+  }
+
+  const LeftSideRightBorderCoordinateValue = 145;
```
Comment (DevBastiS) [inline Version2/index.js:41]: Remove capital letters from variables.

**Review by DevBastiS — CHANGES_REQUESTED** [review]

**Comment by DevBastiS** [comment]
Everything still open

File: Version2/index.js [hunk 50]
```diff
@@ -0,0 +1,163 @@
+let buttonToChase = document.getElementById("button");
```
Comment (stefanluth) [inline Version2/index.js:1]: This can be a `const` as well

File: Version2/index.js [hunk 51]
```diff
@@ -0,0 +1,163 @@
...
+    horizontalDistanceFromButton >= topLeftSideRightBorderCoordinateValue &&
+    horizontalDistanceFromButton <= topLeftSideLeftBorderCoordinateValue &&
+    verticalDistanceFromButton <= topLeftSideUpperBorderCoordinateValue &&
+    verticalDistanceFromButton >= topLeftSideLowerBorderCoordinateValue;
+  if (mouseComesFromTopLeftSide) {
+    setButtonPosition(buttonBox.x - 100 * 5, buttonBox.y - 100 * 3);
+  }
+
+  let xborder = buttonToChase.getBoundingClientRect().x;
+  let yborder = buttonToChase.getBoundingClientRect().y;
```
Comment (stefanluth) [inline Version2/index.js:140]:
```suggestion
  const buttonBorderX = buttonToChase.getBoundingClientRect().x;
  const buttonBorderY = buttonToChase.getBoundingClientRect().y;
```

File: Version2/index.js [hunk 52]
```diff
@@ -0,0 +1,163 @@
...
+  if (buttonHitsAnyEndOfTheBrowser) {
+    buttonToChase.style.left = "250px";
+    buttonToChase.style.top = "70px";
+  }
+});
+
+function setButtonPosition(left, top) {
+  buttonToChase.style.left = `${left}px`;
+  buttonToChase.style.top = `${top}px`;
+}
```
Comment (stefanluth) [inline Version2/index.js:159]: 👍 A text-book perfect function. Short and precise, no side effects, does what it says it should do. More of these :)

**Review by stefanluth — CHANGES_REQUESTED** [review]
Remove .DS_Store :)

**Review by DevBastiS — APPROVED** [review]

**Review by stefanluth — APPROVED** [review]

**Pull request merged**